src/main/java/tech/gruppone/stalker/server/services/LoginService.java

```diff
@@ -0,0 +1,37 @@
+package tech.gruppone.stalker.server.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import reactor.core.publisher.Mono;
+import tech.gruppone.stalker.server.exceptions.LoginFailedException;
+import tech.gruppone.stalker.server.model.api.EncodedJwtDto;
+import tech.gruppone.stalker.server.model.api.LoginDataDto;
+import tech.gruppone.stalker.server.model.db.UserDao;
+import tech.gruppone.stalker.server.repositories.UserRepository;
+import tech.gruppone.stalker.server.security.JwtConfiguration;
+
+@Service
+public class LoginService {
+
+  @Autowired private UserRepository userRepository;
+  @Autowired private JwtConfiguration jwtConfiguration;
```

Comment:
io invece che mettere l'autowired faccio i campi finali e gli do un costruttore con lombok. è più efficiente perché java tratta diversamente i campi `final`!

```suggestion
@AllArgsConstructor
// this makes all fields private and final by default!
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
@Service
public class LoginService {

  UserRepository userRepository;
  JwtConfiguration jwtConfiguration;
```